extensions/amp-accordion/1.0/amp-accordion.js

```diff
@@ -98,20 +102,46 @@ function getState(element, mu) {
       CONTENT_SHIM_PROP,
       bindContentShimToElement
     );
-    const expanded = section.hasAttribute('expanded');
+    const expandStateShim = memo(
+      section,
+      EXPAND_STATE_SHIM_PROP,
+      getExpandStateTrigger.bind(null, element, ActionTrust.HIGH)
+    );
     const props = dict({
       'key': section,
       'as': sectionShim,
       'headerAs': headerShim,
       'contentAs': contentShim,
-      'expanded': expanded,
+      'expanded': section.hasAttribute('expanded'),
       'id': section.getAttribute('id'),
+      'onExpandStateChange': expandStateShim,
     });
     return <AccordionSection {...props} />;
   });
   return dict({'children': children});
 }
 
+/**
+ * @param {!Element} element
+ * @param {!ActionTrust} trust
+ * @param {!Element} section
+ * @return {Function}
+ */
+function getExpandStateTrigger(element, trust, section) {
+  const action = Services.actionServiceForDoc(element);
+  const triggerEvent = (expanded) => {
+    const eventName = expanded ? 'expand' : 'collapse';
+    const event = createCustomEvent(
+      toWin(element.ownerDocument.defaultView),
+      `accordionSection.${eventName}`,
+      dict({})
+    );
+    action.trigger(section, eventName, event, trust);
+    element.dispatchCustomEvent(name);
```

Comment:
We can just do `section.parentElement.dispatchCustomEvent` so we don't need to depend on the `element` in the args.

Reply:
Thanks, updated!